tools/src/main/scala/scala/scalanative/codegen/Lower.scala

```diff
@@ -154,18 +170,33 @@ object Lower {
 
       buf ++= handlers
 
-      eliminateDeadCode(buf.toSeq.map(super.onInst))
+      eliminateDeadCode {
+        buf.toSeq.map(super.onInst)
+      }
     }
 
     override def onVal(value: Val): Val = value match {
```

Comment:
Is method still called from anywhere in practice? If not, it should unconditionally throw to detect problems early. If yes, why is not an issue that it doesn't support `Val.ClassOf` for the places that still call it?

Reply:
All instances of `Val.ClassOf` are being handled before this point with `genVal`.
I've tried to test it against `Inst.If` which is not being handled directly, for expr `if(???) classOf[Int] else classOf[String]` it compiles without problem which means it was handled before `super.onInst`.
Same result when using `match`